src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotHaveIncorrectSuffixTests.cs

```diff
@@ -2015,6 +2015,79 @@ public async Task CA1711_Basic_Diagnostic_FieldNameEndsWithEx()
                     "MyFieldEx"));
         }
 
+        [Fact]
+        public async Task CA1711_EnumFlagSuffix_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public enum SomeEnumFlag
+{
+    X,
+}",
+                GetCSharpResultAt(2, 13, IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.TypeNoAlternateRule, "SomeEnumFlag", IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.FlagSuffix));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Enum SomeEnumFlag
+    X
+End Enum",
+                GetCSharpResultAt(2, 13, IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.TypeNoAlternateRule, "SomeEnumFlag", IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.FlagSuffix));
+        }
+
+        [Fact]
+        public async Task CA1711_EnumFlagsSuffix_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public enum SomeEnumFlags
+{
+    X,
+}",
+                GetCSharpResultAt(2, 13, IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.TypeNoAlternateRule, "SomeEnumFlags", IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.FlagsSuffix));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Enum SomeEnumFlags
+    X
+End Enum",
+                GetCSharpResultAt(2, 13, IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.TypeNoAlternateRule, "SomeEnumFlags", IdentifiersShouldNotHaveIncorrectSuffixAnalyzer.FlagsSuffix));
+        }
+
+        [Fact]
+        public async Task CA1711_AllowedSuffixes_NoDiagnostic()
```

Comment:
I have added only one test for this new option, let me know if you want some specific tests.